bin/init_exercise.py

```diff
@@ -0,0 +1,411 @@
+                indent=2,
+            )
+
+
+@contextmanager
+def git_master(git_path):
+    "A context inside of which you are on the clean master branch"
+    with inside(git_path):
+        dirty = len(output_of('git status --porcelain')) > 0
+        if dirty:
+            subprocess.run(['git', 'stash'])
+        branch = output_of('git rev-parse --abbrev-ref HEAD')
+        if branch != 'master':
+            subprocess.run(['git', 'checkout', 'master'])
+        subprocess.run(['git', 'pull'])
+
+        try:
+            yield
+        finally:
+            if branch != 'master':
+                subprocess.run(['git', 'checkout', branch])
+            if dirty:
+                subprocess.run(['git', 'stash', 'pop'])
+
+
+def generate_readme(exercise_name, get_problem_specification):
+    configlet = None
+    with inside(os.path.join(REPO_ROOT, 'bin')):
+        if not os.path.exists('configlet') and not os.path.exists('configlet.exe'):
+            with inside(REPO_ROOT):
+                subprocess.run('fetch-configlet')
+        for configlet_name in ('configlet', 'configlet.exe'):
+            if os.path.exists(configlet_name):
+                configlet = configlet_name
+                break
+        if configlet is None:
+            print("Could not locate configlet; aborting", file=sys.stderr)
+            sys.exit(1)
+    if get_problem_specification:
+        with inside(os.path.join(REPO_ROOT, '..')):
+            if os.path.exists('problem-specifications'):
+                with git_master('problem-specifications'):
+                    with inside(REPO_ROOT):
+                        subprocess.run([
+                            os.path.join('bin', configlet),
+                            'generate', '.',
+                            '--only', exercise_name,
+                            '--spec-path',
+                            os.path.join('..', 'problem-specifications')
+                        ])
+            else:
+                subprocess.run(
+                    ['git', 'clone', 'https://github.com/exercism/problem-specifications.git']
+                )
+                with inside(REPO_ROOT):
+                    subprocess.run([
+                        os.path.join('bin', configlet),
+                        'generate', '.',
+                        '--only', exercise_name,
+                        '--spec-path',
+                        os.path.join('..', 'problem-specifications')
+                    ])
+    else:
+        with inside(REPO_ROOT):
+            subprocess.run([
+                os.path.join('bin', configlet),
+                'generate', '.',
+                '--only', exercise_name,
+            ])
+
+
+if __name__ == '__main__':
+    import argparse
+    parser = argparse.ArgumentParser(description='Create a Rust Track exercise for Exercism')
+    parser.add_argument('name', help='name of the exercise to create')
+    parser.add_argument('--dont-create-exercise', action='store_true',

```

Comment:
I am used to seeing `--create-exercise` and `--no-create-exercise`, but I don't care and it might not be the case that this is an accepted convention.

Reply:
I haven't seen any other tools of this type; this was all ab nihilo. 

I feel that a verb in the argument is more expressive than simple negation.